### DISCUSSION TITLE
Heat pump for cooling and heating

### DISCUSSION THREAD

**Discussion by sommsmartin** [oemof/tespy#492] [Q&A]
Dear ladies and gentlemen,
i would love to use Tespy for the creation of a heat pump that is able to cool and heat. As far as i can see, the heat pumps described are only meant for heating. Has anybody done that before? If so, would it be possible to share the code of such a heat pump?
Best regards,
Martin

**Comment by fwitte** [comment]
Hi Martin,

in principle you can build that. The main challenge is the data and the modeling approach of the heat exchangers. Do you have information on the surface areas of the heat exchangers and the respective heat transfer coefficients in cooling and heating mode? With those information you could build a tespy model and run simulations in either mode.

If you are only interested in the energy balances you could just build two different models... It depends a bit on what your use case is exactly. Can you elaborate a bit?

Best

**Comment by sommsmartin** [comment]
The problem is being discussed in another Q&A
